src/tests/ftest/util/dmg_utils.py

```diff
@@ -848,6 +895,35 @@ def pool_evict(self, pool):
         """
         return self._get_result(("pool", "evict"), pool=pool)
 
+    def config_generate(self, access_points=None, num_engines=None,
+                        min_ssds=None, net_class=None):
+        """Produce a server configuration.
+        Args:
+            access_points (str): Comma separated list of access point addresses.
+                Defaults to None.
+            num_pmem (int): Number of SCM (pmem) devices required per
+                storage host in DAOS system. Defaults to None.
+            num_nvme (int): Minimum number of NVMe devices required per storage
+                host in DAOS system. Defaults to None.
+            net_class (str): Network class preferred. Defaults to None.
+                i.e. "best-available"|"ethernet"|"infiniband"
+        Returns:
+            dict: the contents of the generate config file.
+        Raises:
+            CommandFailure: if the dmg config generate command fails or if YAML
+                parser encounters an error condition while parsing the contents.
+        """
+        result = self._get_result(
+            ("config", "generate"), access_points=access_points,
+            num_engines=num_engines, min_ssds=min_ssds, net_class=net_class)
+
+        try:
+            yaml_data = yaml.safe_load(result.stdout)
+        except yaml.YAMLError as error:
+            raise CommandFailure(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword